### PR TITLE
Dmp 1737 arm pull response files fix row locking

### PR DESCRIPTION
**Pull request by karen-hedges**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DMP-1737

### Change description ###

Fixed issue whereby the rows were being locked by the transsctional updating most of the EOD table

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmResponseFilesProcessorImpl.java [hunk 1]
```diff
@@ -64,23 +65,29 @@ public class ArmResponseFilesProcessorImpl implements ArmResponseFilesProcessor
...
-
         List<ExternalObjectDirectoryEntity> dataSentToArm =
             externalObjectDirectoryRepository.findByExternalLocationTypeAndObjectStatus(armLocation, armDropZoneStatus);
         if (!CollectionUtils.isEmpty(dataSentToArm)) {
             log.info("ARM Response process found : {} records to be processed", dataSentToArm.size());
             for (ExternalObjectDirectoryEntity externalObjectDirectory : dataSentToArm) {
                 updateExternalObjectDirectoryStatus(externalObjectDirectory, armProcessingResponseFilesStatus);
             }
+            int row = 1;
             for (ExternalObjectDirectoryEntity externalObjectDirectory : dataSentToArm) {
```
Comment (mestebanez) [inline src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmResponseFilesProcessorImpl.java:89]: can we use a more transitional for loop to maintain the count

**Review by cmagowan-hmcts — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmResponseFilesProcessorImpl.java [hunk 2]
```diff
@@ -93,8 +100,18 @@ public void processResponseFiles() {
         }
     }
 
+    @SuppressWarnings("java:S3655")
+    private void initialisePreloadedObjects() {
+        armDropZoneStatus = objectRecordStatusRepository.findById(ARM_DROP_ZONE.getId()).get();
```
Comment (mestebanez) [inline src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmResponseFilesProcessorImpl.java:105]: Could do this with a post constuct annotation?

File: src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmResponseFilesProcessorImpl.java (hunk 1 above)
Comment (karen-hedges) [reply src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmResponseFilesProcessorImpl.java:89]: I understand what you are saying but the row number is just being used for logging purposes. If you change it to a for (int row=0; row < dataSentToArm.size(); row++) you always risk the chance you end up skipping an element as the row number can be changed.

File: src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmResponseFilesProcessorImpl.java (hunk 2 above)
Comment (karen-hedges) [reply src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmResponseFilesProcessorImpl.java:105]: The automated tasks can run at anytime so we want to get the reference objects each time a new task is run

**Pull request merged**